external/fv3fit/fv3fit/reservoir/model.py

```diff
@@ -98,6 +99,105 @@ def load(cls, path: str) -> "HybridReservoirComputingModel":
         )
 
 
+class HybridDatasetAdapter:
+    def __init__(self, model: HybridReservoirComputingModel) -> None:
+        self.model = model
+        self._input_feature_sizes: Optional[Sequence] = None
+
+    def predict(self, inputs: xr.Dataset) -> xr.Dataset:
+        # TODO: centralize stacking logic for encoding decoding
+        # TODO: potentially use in train.py instead of special functions there
+        processed_inputs = self._input_data_to_array(inputs)  # x, y, feature dims
+        if self.model.rank_divider is None:
```

Comment:
This check can be removed now that the rank_divider is a required arg for the models.